### ISSUE TITLE
[BUG] Unable to assign disk when it contains a gpt label

### ISSUE THREAD

**Issue by alexdepalex** [harvester/harvester#1734]
**Describe the bug**
When a disk contains a gpt label, it's not possible to assign it as a data disk. The disk is in the host disk drop-down menu, but it's not selectable.

**To Reproduce**
 Create a disk with PTTYPE="gpt"

**Expected behavior**
Select the disk and purge any existing data that's on there.

**Support bundle**

**Environment:**
 - Harvester ISO version:  v1.0.0
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

```
/dev/nvme2n1: PTUUID="0357b080-149a-4979-b8d1-c6f4e4b9c8c6" PTTYPE="gpt"
/dev/nvme1n1: PTUUID="0ec88787-1cc2-436f-8221-360ecb6a32a4" PTTYPE="gpt"
/dev/nvme0n1: PTUUID="9e215e3b-9111-4f4c-9bc5-dd843821a654" PTTYPE="gpt"
/dev/nvme3n1p1: UUID="ec74890a-78d5-4dfc-b20c-628b92924afc" BLOCK_SIZE="4096" TYPE="ext4" PARTLABEL="primary" PARTUUID="986fdbb9-5e26-4f4b-81d8-7f43d5c35034"
/dev/nvme4n1: PTUUID="4fa299b0-028b-462d-b766-f68757944ba4" PTTYPE="gpt"
/dev/nvme5n1: PTUUID="df114c04-fd3f-493d-ba65-274d2dcd4ddb" PTTYPE="gpt"
```
```
fdisk -l /dev/nvme2n1
Disk /dev/nvme2n1: 1.46 TiB, 1600321314816 bytes, 3125627568 sectors
Disk model: Dell Express Flash PM1725b 1.6TB SFF
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: 0357B080-149A-4979-B8D1-C6F4E4B9C8C6
```

![image](https://user-images.githubusercontent.com/713019/147097622-cb65da1e-0a6f-4dbf-8926-5a876c7b0a2a.png)

```
time="2021-12-22T13:09:03Z" level=info msg="Scan block devices of node: node1"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme0c0n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme0c0n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme0c0n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme1c1n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme1c1n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme1c1n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme2c2n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme2c2n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme2c2n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme3c3n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme3c3n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme3c3n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme4c4n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme4c4n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme4c4n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme5c5n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme5c5n1 : exit status 2\n"
time="2021-12-22T13:09:03Z" level=warning msg="failed to read disk uuid of /dev/nvme5c5n1 : exit status 2\n"
time="2021-12-22T13:09:04Z" level=warning msg="failed to generate GUID for device nvme0c0n1"
time="2021-12-22T13:09:04Z" level=info msg="Skip adding non-identifiable block device /dev/nvme0c0n1"
time="2021-12-22T13:09:04Z" level=warning msg="failed to generate GUID for device nvme1c1n1"
time="2021-12-22T13:09:04Z" level=info msg="Skip adding non-identifiable block device /dev/nvme1c1n1"
time="2021-12-22T13:09:04Z" level=warning msg="failed to generate GUID for device nvme2c2n1"
time="2021-12-22T13:09:04Z" level=info msg="Skip adding non-identifiable block device /dev/nvme2c2n1"
time="2021-12-22T13:09:04Z" level=warning msg="failed to generate GUID for device nvme3c3n1"
time="2021-12-22T13:09:04Z" level=info msg="Skip adding non-identifiable block device /dev/nvme3c3n1"
time="2021-12-22T13:09:04Z" level=warning msg="failed to generate GUID for device nvme4c4n1"
time="2021-12-22T13:09:04Z" level=info msg="Skip adding non-identifiable block device /dev/nvme4c4n1"
time="2021-12-22T13:09:04Z" level=warning msg="failed to generate GUID for device nvme5c5n1"
time="2021-12-22T13:09:04Z" level=info msg="Skip adding non-identifiable block device /dev/nvme5c5n1"
```

**Comment by weihanglo** [comment]
Sorry for the late reply.

Unfortunately I cannot reproduce on my machine. Were those disk once controlled by Harvester before? If it is still possible, can you generate a support bundle? Thanks!

**Comment by weihanglo** [comment]
I guess I can reproduce it with the following steps https://github.com/harvester/harvester/issues/1755#issuecomment-1015204213. Do you have another node with the similar disk layout and they've already partitioned/provisioned? If yes, then this should be the same UI issue.

**Comment by ravarga** [comment]
Reproducer is fairly easy:
1. Deploy a first harvester node with an extra disk (I have used nvme)
2. Deploy and add second node with an extra disk and no partition on top. Such disk will be grayed out and not be possible to add.
3. Same happens on a third node.

The workaround is to create partition on extra disk. Such disk can be selected and added. However it cannot be removed. It gets stuck with "unprovisioning" flag forever.

**Comment by TachunLin** [comment]
Verified **fixed** on master-184241d1-head (02/11). Close this issue.

### Result
Can assign extra disk with and without gpt label to node 2 and node 3 correctly. 

* We can add extra disk `without` **gpt** label to node 3 

  ![image](https://user-images.githubusercontent.com/29251855/153582554-2147cc61-f0ce-49f2-b2ee-94545cf0cb43.png)
  
  ![image](https://user-images.githubusercontent.com/29251855/153582768-10196d1c-1c0f-4a98-9bbd-5159b86828a9.png)
  
  ```
  harvester-node03-220211:~ # fdisk -l /dev/nvme1n1 
  Disk /dev/nvme1n1: 15 GiB, 16106127360 bytes, 31457280 sectors
  Disk model: QEMU NVMe Ctrl                          
  Units: sectors of 1 * 512 = 512 bytes
  Sector size (logical/physical): 512 bytes / 512 bytes
  I/O size (minimum/optimal): 512 bytes / 512 bytes
  harvester-node03-220211:~ # fdisk -l /dev/nvme2n1
  Disk /dev/nvme2n1: 25 GiB, 26843545600 bytes, 52428800 sectors
  Disk model: QEMU NVMe Ctrl                          
  Units: sectors of 1 * 512 = 512 bytes
  Sector size (logical/physical): 512 bytes / 512 bytes
  I/O size (minimum/optimal): 512 bytes / 512 bytes
  ```

* We can also add extra disk `with` **gpt** label on node2 

  ```
  harveter-node02-220211:~ # fdisk -l /dev/nvme0n1
  Disk /dev/nvme0n1: 20 GiB, 21474836480 bytes, 41943040 sectors
  Disk model: QEMU NVMe Ctrl                          
  Units: sectors of 1 * 512 = 512 bytes
  Sector size (logical/physical): 512 bytes / 512 bytes
  I/O size (minimum/optimal): 512 bytes / 512 bytes
  Disklabel type: gpt
  Disk identifier: CF9AFE3A-9F52-45C2-9764-800AE8A8659A
  
  Device         Start      End  Sectors Size Type
  /dev/nvme0n1p1  2048 41940991 41938944  20G Linux filesystem
  ```
  
  ![image](https://user-images.githubusercontent.com/29251855/153576015-fd4ba406-20d8-405a-ad51-62bb6b76cf2a.png)
  
  ![image](https://user-images.githubusercontent.com/29251855/153577253-01399bb4-55c0-488f-b548-3089cd788c5a.png)

### Test Information
* Test Environment: 3 nodes harvester on local kvm machine
* Harvester version: master-184241d1-head (02/11)

### Verify Steps

1. Follow the reproduce steps in #1414 to add extra nvme disk on node 1 ~ node 3 
2. Install harvester node 1 with extra disk 
3. Run `fdisk -l /dev/nvmen1p1` to check gpt label 
4. Check can add extra disk with gpt label to node 1
5. Install harvester node 2 with extra disk
6. Run `fdisk -l /dev/nvmen1p1` to check gpt label
7. Check can add extra disk with gpt label to node 2
8. Install harvester node 3 with extra disk
9. Run `fdisk -l /dev/nvmen1p1` to check gpt label
10. Shutdown node 3 machine, add two additional nvme disk (/dev/nvme1n1, /dev/nvme2n1)
11.  Run `fdisk -l /dev/nvme1n1` to check these two disk have no gpt partition
12. Check can add these two extra disk without gpt partition to node 3
